### PR TITLE
Issue #8 send live location

### PR DESCRIPTION
**Pull request by Suryansh1720001**
@soumenkp2 
Related to Issue #8 
**Pull Request Description: Enhanced Emergency Location Sharing**

- **Achievements:**
  - Checked location permissions.
  - Retrieved user's location using LocationManager.
  - Generated Google Maps link: `https://www.google.com/maps?q=latitude,longitude`.
  - Sent location link via SMS using SmsManager.
  - Informed user with user-friendly Toast notifications.

- **Considerations:**
  - Handled runtime location permission requests.
  - Implemented robust error handling for location retrieval.
  - Ethically implemented, considering emergency context and privacy laws.

- **Testing:**
  - Verified functionality on various devices and Android versions.
  - Ensured accurate opening of Google Maps link.

**Note:** Please review and merge for an improved emergency section, providing a reliable and user-friendly location-sharing feature.

### CODE REVIEW

**Comment by Suryansh1720001** [comment]
Hello @soumenkp2,

I believe this issue requires a hard difficulty level. I have invested significant effort in successfully implementing it and have added additional functions. Now, if the location is not successfully fetched, dialog boxes will appear to handle different scenario.

**Comment by soumenkp2** [comment]
No issues @Suryansh1720001 
I would be happy to label this as hard level. Let me know once you can debug the issue that I have sent to you on Discord so we can move ahead to merge this down.

**Comment by soumenkp2** [comment]
Hi @Suryansh1720001 , 
I am seeking an update on the current issue.

**Comment by Suryansh1720001** [comment]
@soumenkp2 
Actually, my exams are currently ongoing, which was why I haven't been able to work on it. 
However, you can now test it on your mobile - I have committed the changes. I checked it on an emulator with Android 10, and it works fine on that.

File: app/src/main/java/com/example/pehchaan/fragment/EmergencyFragment.java [hunk 1]
```diff
@@ -244,14 +266,22 @@ public void onClick(View view) {
...
+                    EmergencyContactNumber.set(2, num3.getString("num3", "Enter Num3"));
+                    EmergencyContactNumber.set(3, num4.getString("num4", "Enter Num4"));
+                }
                 CustomEmergencyMess = nummsg.getString("nummsg", "Enter your message");
 
+                locationRequest();
+
+
                 /*Intent sms1 = new Intent(Intent.ACTION_VIEW);
                 sms1.setData(Uri.parse("smsto:"));
```
Comment (soumenkp2) [inline app/src/main/java/com/example/pehchaan/fragment/EmergencyFragment.java]: do we need this commented code?

File: app/src/main/java/com/example/pehchaan/fragment/EmergencyFragment.java [hunk 2]
```diff
@@ -1,27 +1,43 @@
...
-
+import com.example.pehchaan.utils.LocationManagerHelper;
 import java.util.ArrayList;
 import java.util.List;
 
+
 public class EmergencyFragment extends Fragment {
 
+    private static final int LOCATION_PERMISSION_REQUEST_CODE = 1;
+    //    private static final int LOCATION_SETTINGS_REQUEST_CODE = 2;
```
Comment (soumenkp2) [inline app/src/main/java/com/example/pehchaan/fragment/EmergencyFragment.java]: remove this commented code

**Comment by soumenkp2** [comment]
Do resolve the comments please and as requested I will be closing this issue as 'hard' labeled. 
Good work! 
Thanks for the contribution.

File: app/src/main/java/com/example/pehchaan/fragment/EmergencyFragment.java (hunk 1 above)
Comment (Suryansh1720001) [reply app/src/main/java/com/example/pehchaan/fragment/EmergencyFragment.java]: Actually, it was already present; that's why I didn't remove it. But it is not necessary.

File: app/src/main/java/com/example/pehchaan/fragment/EmergencyFragment.java (hunk 2 above)
Comment (Suryansh1720001) [reply app/src/main/java/com/example/pehchaan/fragment/EmergencyFragment.java]: ok

**Comment by Suryansh1720001** [comment]
@soumenkp2 
Now check!

**Pull request merged**